ODIN_II/SRC/ast_util.cpp

```diff
@@ -1086,6 +1091,9 @@ static void check_binary_operation(ast_node_t **node)
  *-------------------------------------------------------------------------------------------*/
 ast_node_t *fold_unary(ast_node_t *node)
 {
+	if(!node)
```

Comment:
if you're gonna check this on fold_unary, you should check it on fold_binary too please

Reply:
I've made both function more consistent with each other. last commit should resolve your comment